xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -1636,10 +1636,22 @@ bool CGUIDialogVideoInfo::GetSetForMovie(const CFileItem *movieItem, CFileItemPt
   if (!videodb.Open())
     return false;
 
+  const std::shared_ptr<CSettings> settings = CServiceBroker::GetSettingsComponent()->GetSettings();
+  const bool m_groupSingleItemSets =
+      settings->GetBool(CSettings::SETTING_VIDEOLIBRARY_GROUPSINGLEITEMSETS);
+
+  // if "Include sets containing a single movie" is disabled in settings
+  // we need to temporary enable it while fetching the movieset list ...
+  if (!m_groupSingleItemSets)
+    settings->SetBool(CSettings::SETTING_VIDEOLIBRARY_GROUPSINGLEITEMSETS, true);
+
   CFileItemList listItems;
-  std::string baseDir = "videodb://movies/sets/";
-  if (!CDirectory::GetDirectory(baseDir, listItems, "", DIR_FLAG_DEFAULTS))
-    return false;
+  CDirectory::GetDirectory("videodb://movies/sets/", listItems, "", DIR_FLAG_DEFAULTS);
```

Comment:
removing the exit-code check had no negative effects with an emptied „sets“-table